### ISSUE TITLE
cfg_attr + no_main causes false warning

### ISSUE THREAD

**Issue by CryZe** [rust-lang/rust#51141]
If you conditionally use the no_main attribute, you get a warning:

src/lib.rs
```rust
#![cfg_attr(target_os = "emscripten", no_main)]
...
```
and then compile it as
```
cargo build --target asmjs-unknown-emscripten
```
results in the following warning:
```
warning: crate-level attribute should be in the root module
 --> capi/src/lib.rs:1:39
  |
1 | #![cfg_attr(target_os = "emscripten", no_main)]
  |                                       ^^^^^^^^
```

**Comment by shymega** [comment]
@CryZe Couldn't help noticing you closed this issue. I'm having a similar issue with a kernel I'm working on - it doesn't use conditional compilation, but also has the same warning. Did you find a way to rectify the problem? Cheers.

**Comment by CryZe** [comment]
Pretty sure in my case it was because I accidentally used lib.rs instead of main.rs.

**Comment by shymega** [comment]
Thanks for the reply. I was trying to do the same thing, but it doesn't appear to work that way.
